ncc/examples/basic.c

```diff
@@ -0,0 +1,1929 @@
+//
+// UVM Basic
+// by Abdul Bahajaj
+// 
+// Note to reader: When I authored this VM there were a lot of missing features from the NCC compiler ( e.g. there are no structs, unions, etc...).
+// Keeping this in mind might help you understand why the code is structured in the way it is.
+
+
+#include <stddef.h>
+#include <string.h>
+#include <uvm/syscalls.h>
+#include <uvm/utils.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <uvm/graphics.h>
+
+#define FONT_MONOGRAM_NUMBER_OF_CHARACTERS 390
+#define FONT_MONOGRAM_HEIGHT 12
+#define FONT_MONOGRAM_WIDTH 7
+
+
+#define FRAME_WIDTH 1200
+#define FRAME_HEIGHT 700
+#define NUM_COLS 25
+#define NUM_ROWS 21
+
+
+#ifdef DEBUG
+#define DEBUGS(s) puts(__FILE__ " : "); print_i64(__LINE__); puts(" "); puts(s);
+#define DEBUGI(i) print_i64(i);
+#else 
+#define DEBUGS(s) 
+#define DEBUGI(i) 
+#endif
+
+#define DEBUG(s) DEBUGS(s "\n");
+
+#define TRY(exp) if(exp) return 1;
+#define NULLGAURD(ptr) if(ptr == NULL) return NULL;
+
+
+size_t console_width;
+
+size_t margin = 5;
+size_t char_width = 18;
+size_t char_height = 32;
+
+u32 frame_buffer[FRAME_HEIGHT][FRAME_WIDTH];
+
+char text[NUM_ROWS][NUM_COLS];
+
+
+// Position of the cursor
+size_t line_idx = 0;
+size_t min_line_idx = 4;
+size_t col_idx = 0;
+
+size_t row_len(size_t row_idx)
+{
+	for (int i = 0; i < NUM_COLS; ++i)
+	{
+		if (text[row_idx][i] == 0)
+		{
+			return i;
+		}
+	}
+
+	return NUM_COLS;
+}
```

Comment:
Could I ask you to convert this to 4 spaces per indentation, no tabs?

Reply:
yes on it.

Reply:
done